### PR TITLE
[feat] MobX Upgrade (again)

### PR DESCRIPTION
**Pull request by visormatt**
# Summary

This PR updates the Survey to current MobX package version. We've taken the [codemod](https://www.npmjs.com/package/mobx-undecorate) approach and removed all decorators. 

**Resources:**

- [MobX Migration Guide](https://mobx.js.org/migrating-from-4-or-5.html#upgrading-classes-to-use-makeobservable)
- [mobx-undecorate codemod](https://www.npmjs.com/package/mobx-undecorate)
- [StackOverflow](https://stackoverflow.com/questions/48437102/mobx-why-should-i-use-observer-when-i-could-use-inject-when-injecting-data)
  - [Related Github ticket](https://github.com/mobxjs/mobx-react/issues/410)

## Testing

- [ ] Ignore all the changes in `dist` this is compiled code
- [ ] Review the changes in the `src` files
  - There is noise due to formatting, necessary evil
  - Decorators have been removed in favor `makeObservable` _(codemod)_
- [ ] Testing the survey will need to happen in the [NativeApp PR](https://github.com/shiftsmartinc/nativeapps/pull/741)

### CODE REVIEW

File: src/Model.ts [hunk 1]
```diff
@@ -27,61 +27,67 @@ const randomizeArray = (array: Array<string>) => {
...
   }
   return array;
-}
+};
 
 class Page {
   collection;
   name;
-  @observable _visible;
+  _visible = true;
```
Comment (visormatt) [inline src/Model.ts:35]: We needed a default value here

File: src/Model.ts [hunk 2]
```diff
@@ -27,61 +27,67 @@ const randomizeArray = (array: Array<string>) => {
...
 class Question {
-  @observable visible;
-  @observable value = null;
-  @observable error = null;
-  @observable comment = null;
-  @observable number;
-  @observable questions = [];
-  @observable choices = [];
-  @observable title = '';
+  visible = false;
```
Comment (visormatt) [inline src/Model.ts:83]: Needed a default here as well

File: src/QuestionRadioItem.tsx [hunk 3]
```diff
@@ -70,32 +70,44 @@ export interface Props {
...
+          />
+        )}
       </TouchableWithFeedback>
     );
   }
 }
+
+export default inject((store: any) => ({
+  isPreview: store.model.isPreview,
+}))(observer(QuestionRadioItem));
```
Comment (visormatt) [inline src/QuestionRadioItem.tsx:113]: This is the recommended approach (not using decorators) which fixed the rendering issue.

**Review by angelo-hub — APPROVED** [review]
LGTM

**Pull request merged**